algorithms/benches/snark/marlin.rs

```diff
@@ -19,55 +19,21 @@ extern crate criterion;
 
 use snarkvm_algorithms::{
     crypto_hash::PoseidonSponge,
-    snark::marlin::{ahp::AHPForR1CS, CircuitVerifyingKey, MarlinHidingMode, MarlinSNARK},
+    snark::marlin::{ahp::AHPForR1CS, CircuitVerifyingKey, MarlinHidingMode, MarlinSNARK, TestCircuit},
```

Comment:
Hm should we be exposing the `TestCircuit` from `algorithms`?

Reply:
Do you mean:
1. it should perhaps be moved to e.g. `utilities` (which I think not, because it is not used outside of `algorithms`)
2. it is not scoped properly (which might be true, I can change `pub` to `pub(crate)`

Reply:
Right, I meant that it's `pub` in `algorithms`, when it ideally shouldn't be. However I'm pretty sure that you can't mark it as `pub(crate)`, since benchmarks use the library as an outside consumer.

I think the cleanest solution is to leave `TestCircuit` as `pub`, but to additionally mark it as `#[doc(hidden)]`, and to add documentation very clearly warning the user that `TestCircuit` is only for testing and should not be used in any production code.

Reply:
Done in b17840352